openerp/addons/base/migrations/7.0.1.3/pre-migration.py

```diff
@@ -202,6 +207,31 @@ def create_users_partner(cr):
         "WHERE module='base' and name='user_root'")
     user_root = cr.fetchone()
     user_root_id = user_root and user_root[0] or 0
+
+    if openupgrade.column_exists(
+            cr, 'hr_employee', 'address_id') and\
+            openupgrade.column_exists(
+                cr, 'resource_resource', 'user_id'):
+        # if hr is installed, get partners from there if possible
+        cr.execute(
+            '''UPDATE res_users SET partner_id=u.partner_id
+            FROM (
+                SELECT r.user_id, a.partner_id
+                FROM hr_employee e
+                JOIN resource_resource r ON e.resource_id=r.id
+                JOIN res_partner_address a ON COALESCE(
+                    e.address_id, e.address_home_id)=a.id
```

Comment:
I think the latter is better, because in general we don't want the user's home address.

How about something like 
```
COALESCE(e.address_id, e.address_home_id)=a.id WHERE r.user_id IS NOT NULL AND a.partner_id IS NOT NULL AND a.partner_id <> ${base.main_partner}
```
?